### PR TITLE
Split Razor Components topic

### LINKED ISSUE #16904: Split Razor Components topic
Discussion ...

* https://github.com/aspnet/AspNetCore.Docs/issues/14353#issuecomment-532190780
* https://github.com/aspnet/AspNetCore.Docs/issues/16887#issuecomment-583245467

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: be4c4d0c-149c-bde1-c7fa-f082916390ca
* Version Independent ID: 9d945996-ad02-bd4b-58ea-c465cc5cc74c
* Content: [Create and use ASP.NET Core Razor components](https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.1)
* Content Source: [aspnetcore/blazor/components.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/components.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #16904

[Internal Review (links to the Components topic; navigate the remaining from the sidebar TOC)](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.1&branch=pr-en-us-16987)

* Seems to have split rather cleanly. I was able to turn the _Components **BOOK!**_ into a much shorter, focused topic with split out topics for ...
  * Data binding
  * Event handling
  * Templated components
  * Integrate components (into RP/MVC apps)
  * Globalization and localization
  * Advanced scenarios (Just one right now, *Manual RenderTreeBuilder logic*. We can add more as needed going forward.)
* Examine the ordering in the *toc.yml* file at the end of the diff to see if this sequence makes sense reading these top-to-bottom, as many devs consume node topics in that sequence.
* The new *Integrate components* topic is especially good here given that it combines app-level concerns that were covered in the *Hosting model config* topic with the component-level concerns from the *Components* topic.

### CODE REVIEW

File: aspnetcore/blazor/events.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
+---
+title: ASP.NET Core Blazor events
```
Comment (mkArtakMSFT) [inline aspnetcore/blazor/events.md]: Doing this we should also think about SEO.
Remember, that we have a big goal of making all these doc articles more discoverable. From that perspective, I'm not sure whether people would use `handling` as a phrase to look for in title vs only `Events`. @danroth27 what do you think about this one: `Events` vs `Event Handling`

Comment (guardrex) [reply aspnetcore/blazor/events.md]: That's cool if you feel it best. The docs guidelines only ask for it where possible. For things that are actually gerunds, it's cool with them (e.g., "logging," "data binding," etc.).

Comment (guardrex) [reply aspnetcore/blazor/events.md]: **_Done!_** :+1:

**Comment by guardrex** [comment]
Build issue of some sort ... rebuilding with an open-close of the PR.

**Pull request closed without merging**